plugins/promela/hu.bme.mit.gamma.promela.verification/src/hu/bme/mit/gamma/promela/verification/PromelaVerifier.xtend

```diff
@@ -32,38 +32,41 @@ class PromelaVerifier extends AbstractVerifier {
 	override Result verifyQuery(Object traceability, String parameters, File modelFile, File queryFile) {
 		val model = fileUtil.loadString(modelFile)
 		val query = fileUtil.loadString(queryFile)
-		
-		var modelWithLtls = model
+
 		var i = 0
+		var Result result = null
+		
 		for (singleQuery : query.split(System.lineSeparator).reject[it.nullOrEmpty]) {
 			// Supporting multiple queries in separate files
 			val ltl = '''«System.lineSeparator»ltl ltl_«i» { «singleQuery.adaptQuery» }'''
-			modelWithLtls += ltl
+			val modelWithLtl = model + ltl
 			i++
+			
+			val rootGenFolder = new File(modelFile.parent, "." + fileUtil.getExtensionlessName(modelFile))
+			rootGenFolder.deleteOnExit
+			rootGenFolder.mkdirs
+			// save model with all LTL
+			val tmpGenFolder = new File(rootGenFolder + File.separator + fileUtil.getExtensionlessName(modelFile) + "-LTL" + System.currentTimeMillis.toString)
+			tmpGenFolder.deleteOnExit
```

Comment:
This `deleteOnExit` call works only if the folder is empty. Use `forceDeleteOnExit.forceDeleteOnExit`, and check if it works according to your expectations.